invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -69,9 +71,17 @@ def record_cls(self):
         """Get specified record class or from service."""
         return self._record_cls or self.get_service().record_cls
 
+    @property
+    def draft_cls(self):
+        """Get specified draft class or from service."""
+        return self._draft_cls or self.get_service().draft_cls
+
     def matches_entity(self, entity):
         """Check if the entity is a result item."""
-        return isinstance(entity, (self.item_cls, self.record_cls))
+        try:
+            return isinstance(entity, (self.item_cls, self.record_cls, self.draft_cls))
+        except AttributeError:
```

Comment:
I am a bit worried that attribute level might be a bit too generic and hide other exceptions (not only the ones produced by accessing self.item_cls self.record_cls, self.draft_cls), but I don't have a good alternative for now